### PR TITLE
Adds Generic Error model

### PR DESCRIPTION
**Pull request by Moejoe90**
Also this Connects Generic Error model with Register lightwallet
image

### CODE REVIEW

File: app/components/modals/GenericErrorModal.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+interface IDispatchProps {
+  onCancel: () => void;
+}
+
+const GenericErrorModalComponent: React.SFC<IStateProps & IDispatchProps> = props => (
+  <Modal isOpen={props.isOpen} toggle={props.onCancel}>
+    <ModalBody>
+      <Row className="mt-5 justify-content-center">
```
Comment (banciur) [inline app/components/modals/GenericErrorModal.tsx]: You are using `justify-content-center` which adds flex property. Maybe it makes more sense to add `text-center` to `<ModalBody>`

Comment (Moejoe90) [reply app/components/modals/GenericErrorModal.tsx]: I didn't focus on design much, the main point was to connect everything together now and then focus on presentation and design later 
For example, the cancel button most probably will change into an icon ... etc.

File: app/modules/wallet-selector/light-wizard/flows.ts [hunk 2]
```diff
@@ -101,5 +108,11 @@ function mapLightWalletErrorToErrorMessage(e: Error): string {
   if (e instanceof LightDeserializeError) {
     return "Problem with Vault retrieval";
   }
+  if (e instanceof LightKeyEncryptError) {
+    return "Problem with lightWallet Encryption";
```
Comment (krzkaczor) [inline app/modules/wallet-selector/light-wizard/flows.ts]: light wallet encryption*

File: app/modules/wallet-selector/light-wizard/actions.spec.ts [hunk 3]
```diff
@@ -88,7 +88,7 @@ describe("Wallet selector > Light wallet wizard > actions", () => {
       );
 
       expect(dispatchMock).to.be.calledWithExactly(
-        actions.wallet.lightWalletConnectionError("Password is not correct"),
+        actions.genericErrorModal.showError("LightWallet Error", "Password is not correct"),
```
Comment (krzkaczor) [inline app/modules/wallet-selector/light-wizard/actions.spec.ts]: Light wallet *

File: app/modules/wallet-selector/light-wizard/flows.ts (hunk 2 above)
Comment (krzkaczor) [reply app/modules/wallet-selector/light-wizard/flows.ts]: Do you see a possibility to merge it with errors.ts file? I think i extracted that file by accident but didnt copy all error handling code.

File: app/modules/genericErrorModal/reducer.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { AppReducer } from "../../store";
+
+export interface IGenericErrorModalState {
+  isOpen: boolean;
+  errorObj: IErrorObj;
```
Comment (krzkaczor) [inline app/modules/genericErrorModal/reducer.ts]: this should be optional.

File: app/modules/genericErrorModal/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+}
+export interface IErrorObj {
+  mainError: string;
+  errorMsg: string;
+}
+const initialState: IGenericErrorModalState = {
+  isOpen: false,
+  errorObj: { mainError: "", errorMsg: "" },
```
Comment (krzkaczor) [inline app/modules/genericErrorModal/reducer.ts]: then this is just undefined.

File: app/modules/genericErrorModal/reducer.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    case "GENERIC_ERROR_MODAL_SHOW":
+      return {
+        ...state,
+        isOpen: true,
+        errorObj: action.payload,
+      };
+    case "GENERIC_ERROR_MODAL_HIDE":
+      return {
+        ...state,
+        isOpen: false,
```
Comment (krzkaczor) [inline app/modules/genericErrorModal/reducer.ts]: here it's better to clear errorObj IMO as well

File: app/modules/wallet-selector/light-wizard/flows.ts (hunk 2 above)
Comment (Moejoe90) [reply app/modules/wallet-selector/light-wizard/flows.ts]: Yes absolutely, i had that in mind earlier

File: app/modules/genericErrorModal/reducer.ts (hunk 4 above)
Comment (Moejoe90) [reply app/modules/genericErrorModal/reducer.ts]: Now that its optional an additional check is added

File: app/modules/genericErrorModal/reducer.ts (hunk 5 above)
Comment (Moejoe90) [reply app/modules/genericErrorModal/reducer.ts]: done

File: app/modules/genericErrorModal/reducer.ts (hunk 6 above)
Comment (Moejoe90) [reply app/modules/genericErrorModal/reducer.ts]: cleared

File: app/modules/wallet-selector/light-wizard/actions.spec.ts (hunk 3 above)
Comment (Moejoe90) [reply app/modules/wallet-selector/light-wizard/actions.spec.ts]: done

**Pull request merged**